Lib/asyncio/base_events.py

```
@@ -1016,38 +1016,40 @@ async def _connect_sock(self, exceptions, addr_info, local_addr_infos=None):
-                    if my_exceptions:
-                        raise my_exceptions.pop()
-                    else:
-                        raise OSError(f"no matching local address with {family=} found")
-            await self.sock_connect(sock, address)
-            return sock
-        except OSError as exc:
-            my_exceptions.append(exc)
-            if sock is not None:
-                sock.close()
-            raise
+            try:
+                sock = socket.socket(family=family, type=type_, proto=proto)
+                sock.setblocking(False)
+                if local_addr_infos is not None:
+                    for lfamily, _, _, _, laddr in local_addr_infos:
+                        # skip local addresses of different family
+                        if lfamily != family:
+                            continue
+                        try:
+                            sock.bind(laddr)
+                            break
+                        except OSError as exc:
+                            msg = (
+                                f'error while attempting to bind on '
+                                f'address {laddr!r}: {str(exc).lower()}'
+                            )
+                            exc = OSError(exc.errno, msg)
+                            my_exceptions.append(exc)
+                    else:  # all bind attempts failed
+                        if my_exceptions:
+                            raise my_exceptions.pop()
+                        else:
+                            raise OSError(f"no matching local address with {family=} found")
+                await self.sock_connect(sock, address)
+                return sock
+            except OSError as exc:
+                my_exceptions.append(exc)
+                raise
         except:
             if sock is not None:
-                sock.close()
+                try:
+                    sock.close()
+                except OSError:
```

I am not sure if suppressing this error is correct here.

---

I think it is okay. It is an error in closing just created connection, it is less interesting than the original non-OSError error.

We do not even know what is this error, because we have no reproducer. ECONNRESET is already ignored in `socket.close()`. I suspect that this is similar error (maybe EINTR?) which should always be ignored.

---

Okay, please add a comment about this with linking to issue as this isn't obvious.

---

What comment is needed here? Ignoring any OSError is pretty common, and such code usually does not have comment. If there is a comment (e.g. "Ignore broken pipes caused by quitting the pager program."), it specifies the most common cause of error, but in this case we do not know it.

---

just mention the issue number, no need for elaborate comment

---

There are two ways to solve this issue:
* add this exception to `my_exceptions`
* ignore it.

Otherwise we will get IndexError or, after merging #135845, wrong TimeoutError.

I think that this error is not important.